frontend/www/js/omegaup/arena/arena.test.ts

```diff
@@ -51,5 +53,134 @@ describe('arena', () => {
+            }),
+          });
+        }
+        if (req.url == '/api/problemset/scoreboard/') {
+          return Promise.resolve({
+            status: 200,
+            body: JSON.stringify({
+              status: 'ok',
+              start_time: serverTime,
+              finish_time: serverTime + 3600,
+            }),
+          });
+        }
+        if (req.url == '/api/contest/clarifications/') {
+          return Promise.resolve({
+            status: 200,
+            body: JSON.stringify({
+              status: 'ok',
+              start_time: serverTime,
+              finish_time: serverTime + 3600,
+            }),
+          });
+        }
+        if (req.url == '/api/run/create/') {
+          return Promise.resolve({
+            status: 200,
+            body: JSON.stringify({
+              status: 'ok',
+              nextSubmissionTimestamp: serverTime / 1000 + 60,
+              submission_deadline: serverTime / 1000 + 3600,
+            }),
+          });
+        }
+        if (req.url == '/api/run/status/') {
+          return Promise.resolve({
+            status: 200,
+            body: JSON.stringify({
+              status: 'ok',
+              start_time: serverTime,
+              finish_time: serverTime + 3600,
+            }),
+          });
+        }
+        return Promise.resolve({
+          ok: false,
+          status: 404,
+          body: JSON.stringify({
+            status: 'error',
+            error: `Invalid call to "${req.url}" in test`,
+            errorcode: 403,
+          }),
+        });
+      });
+
+      let dateNowSpy: jest.SpyInstance<number, []> | null = null;
+      dateNowSpy = jest.spyOn(Date, 'now').mockImplementation(() => now);
+      jest.useFakeTimers();
+      const arenaInstance = new arena.Arena(
+        GetOptionsFromLocation(
+          new window.URL('http://localhost:8001/arena/test/'),
+        ),
+      );
+      arenaInstance.problems = {
+        test: {
+          accepts_submissions: true,
+          alias: 'test',
+          commit: 'abcdef',
+          input_limit: 10240,
+          languages: ['py2', 'py3'],
+          points: 0,
+          quality_seal: true,
+          title: 'Test',
+          visibility: 2,
+        },
+      };
+      arenaInstance.currentProblem = {
+        accepts_submissions: true,
+        title: 'Test',
+        alias: 'test',
+        commit: 'abcdef',
+        source: 'omegaUp',
+        languages: ['py2', 'py3'],
+        points: 0,
+        input_limit: 10240,
+        quality_seal: true,
+        visibility: 2,
+      };
+      await arenaInstance.submitRun('print(3)', 'py3');
+      if (arenaInstance.currentProblem.nextSubmissionTimestamp) {
+        expect(
+          (arenaInstance.currentProblem.nextSubmissionTimestamp?.getTime() -
+            serverTime) /
+            1000,
+        ).toEqual(60);
+        expect(
+          (arenaInstance.currentProblem.nextSubmissionTimestamp?.getTime() -
```

Comment:
```suggestion
          (arenaInstance.currentProblem.nextSubmissionTimestamp.getTime() -
            serverTime) /
            1000,
        ).toEqual(60);
        expect(
          (arenaInstance.currentProblem.nextSubmissionTimestamp.getTime() -
```

arriba ya se aseveró que no es nullish, así que `?.` no se debería usar.

Reply:
Ok, se quitan.